service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/amendment/list/AmendableOrderListBuilder.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.fpl.service.orders.amendment.list;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.interfaces.AmendableOrder;
+import uk.gov.hmcts.reform.fpl.service.DynamicListService;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.reverseOrder;
+import static uk.gov.hmcts.reform.fpl.enums.State.CLOSED;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class AmendableOrderListBuilder {
+    private final DynamicListService listService;
+    private final List<AmendableListItemProvider> providers;
+
+    public Optional<DynamicList> buildList(CaseData caseData) {
+        if (CLOSED == caseData.getState()) {
+            return Optional.empty();
+        }
+
+        Comparator<Element<? extends AmendableOrder>> comparator = comparing(
```

Comment:
going to keep it as a field var, if I try to chain the then comparing as I would want to if it was a constant gives an issue where it thinks it is dealing with an `Object` instead of `Element<? extends AmendableOrder>`